Plugins/Encounter/Encounter.cpp

```diff
@@ -192,12 +199,78 @@ ArgumentStack Encounter::SetResetTime(ArgumentStack&& args)
     if (auto *pEncounter = encounter(args))
     {
         auto resetTime = Services::Events::ExtractArgument<int32_t>(args);
-          ASSERT_OR_THROW(resetTime >= 0);
+        ASSERT_OR_THROW(resetTime >= 0);
 
         pEncounter->m_nResetTime = resetTime;
     }
 
     return Services::Events::Arguments();
 }
 
+ArgumentStack Encounter::GetNumberOfSpawnPoints(ArgumentStack&& args)
+{
+    int32_t retVal = 0;
+
+    if (auto *pEncounter = encounter(args))
+    {
+        retVal = pEncounter->m_nNumSpawnPoints;
+    }
+
+    return Services::Events::Arguments(retVal);
+}
+
+ArgumentStack Encounter::GetSpawnPointByIndex(ArgumentStack&& args)
+{
+    float x = 0.0, y = 0.0, z = 0.0, o = 0.0;
+
+    if (auto *pEncounter = encounter(args))
+    {
+        const auto index = Services::Events::ExtractArgument<int32_t>(args);
+        ASSERT_OR_THROW(index >= 0);
+
+        if (index < pEncounter->m_nNumSpawnPoints)
+        {
+            x = pEncounter->m_pSpawnPointList[index].m_vPosition.x;
+            y = pEncounter->m_pSpawnPointList[index].m_vPosition.y;
+            z = pEncounter->m_pSpawnPointList[index].m_vPosition.z;
+            o = pEncounter->m_pSpawnPointList[index].m_fOrientation;
```

Comment:
I wonder if orientation actually does anything here. And if not, you could just return a `vector` instead of `location`. No big deal either way..

Reply:
I have no clue if orientation is used by the NWN engine, but a user using this API can use the orientation if he wishes.